modules/pipeline/providers/leaderworker/leader_framework.go

```diff
@@ -89,6 +89,7 @@ func (p *provider) mergeWithInternalLeaderListeners() []Listener {
 		&DefaultListener{BeforeExecOnLeaderFunc: asyncWrapper(p.leaderListenLogicTaskChange)},
 		&DefaultListener{BeforeExecOnLeaderFunc: asyncWrapper(p.leaderListenTaskCanceling)},
 		&DefaultListener{BeforeExecOnLeaderFunc: asyncWrapper(p.loadCancelingTasks)},
+		&DefaultListener{BeforeExecOnLeaderFunc: asyncWrapper(p.DialerAgent.RegisterEdgeToDialer)},
```

Comment:
why add here

Reply:
adjusted, move to leaderwork on leader
![image](https://user-images.githubusercontent.com/30427474/163778803-b8af4b50-552e-4a42-b8de-16e26854e0c5.png)

Reply:
move to edgepipeline_register provider.run is better.

Reply:
adjusted